### ISSUE TITLE
Misclassification DataFrame is badly composed in LDA notebook

### ISSUE THREAD

**Issue by chrisfalter** [stefan-jansen/machine-learning-for-trading#51]
The week 7 practice notebook 04_lda_with_sklearn.ipynb contains a badly compiled pandas DataFrame. It groups the test set predictions by ground truth topic, _which reorders the rows_. However, it leaves the headings and articles _in their original, ungrouped order_ as it concatenates them into the DataFrame. The rows are thus jumbled chaotically.

```python
test_assignments = test_opt_eval.groupby(level='topic').idxmax(axis=1)
test_assignments = test_assignments.reset_index(-1, drop=True).to_frame('predicted').reset_index()
test_assignments['heading'] = test_docs.heading.values
test_assignments['article'] = test_docs.article.values
test_assignments.head(6)

# Output:
#      topic predicted  heading                           article
# 0 Business Topic 4    Kilroy launches 'Veritas' party   Ex-BBC chat show host and East Midlands MEP R...
# 1 Business Topic 4    Radcliffe eyes hard line on drugs Paula Radcliffe has called for all athletes f...
# 2 Business Topic 4    S Korean consumers spending again South Korea looks set to sustain its revival ...
# 3 Business Topic 4    Quiksilver moves for Rossignol    Shares of Skis Rossignol, the world's largest...
# 4 Business Topic 4    Britons fed up with net service   A survey conducted by PC Pro Magazine has rev...
# 5 Business Topic 4    Scissor Sisters triumph at Brits  US band Scissor Sisters led the winners at th.
```
According to the code's output, the ground truth classification for all 6 articles is "Business." However, the classifications are in fact "Politics," "Sport," "Business," "Business," "Tech," and "Entertainment."

Here's code that fixes the problem:
```python
test_assignments = pd.DataFrame(test_eval.idxmax(axis=1), columns=["prediction"])
test_assignments['heading'] = test_docs.heading.values
test_assignments['article'] = test_docs.article.values
test_assignments.head()

# Output
#    topic   prediction heading                           article
# Politics   Topic 5    Kilroy launches 'Veritas' party   Ex-BBC chat show host and East Midlands MEP R...
# Sport      Topic 3    Radcliffe eyes hard line on drugs Paula Radcliffe has called for all athletes f...
# Business   Topic 2    S Korean consumers spending again South Korea looks set to sustain its revival ...
# Business   Topic 2    Quiksilver moves for Rossignol    Shares of Skis Rossignol, the world's largest...
# Tech       Topic 4    Britons fed up with net service   A survey conducted by PC Pro Magazine has rev...
```
You have my permission to use my code and analysis; I would only request the courtesy of being credited for any contribution I make toward your final product. (Thanks!)

**Comment by stefan-jansen** [comment]
Thank you for pointing out a bug. Usually, I would now go ahead and fix it accordingly. How should I credit you to properly account for your effort? Would a comment in the code be sufficient?

**Comment by chrisfalter** [comment]
Thanks for the response! Yes, a comment in the code would be fine.
